lib/ddtrace/span/external_resource.rb

```diff
@@ -0,0 +1,38 @@
+require 'ddtrace/ext/app_types'
+require 'ddtrace/ext/http'
+
+module Datadog
+  class Span
+    # Extension for Datadog::Span that tracks if this span
+    # represents a unit of work performed by this application
+    # or an external resource being indirectly measured
+    # (e.g. SQL, Redis, outbound HTTP request).
+    module ExternalResource
+      attr_writer :external_resource
+
+      # Span types of known to represent an internal application trace
+      INTERNAL_APPLICATION_SPAN_TYPES = [
```

Comment:
hmm... curious if a blacklist would be better here or not...

(not trying to hint that I think you should do that, actually just pondering it)

Reply:
My thinking was that we have a more external resources than internal (9 vs 4).

I was pondering: "if a new span time comes up, is it likely to be internal or external"? And my educated guess is that it is more likely to be a new external system. Nothing set in stone, though.

Reply:
I think in general this approach works.

Doesn't matter if it is whitelist or blacklist, whether it is only based on the span type or maybe needs to also check operation names, or something else. But for now, this works, we might need to be flexible in the future on what defines an "external resource", but this general pattern/approach works for me.